api/lib/models/location/location-data-mapper-impl.ts

```diff
@@ -0,0 +1,170 @@
+import { MysqlUow } from '../../services/database/svc/mysql-uow.service';
+import { IUowOpts } from '../../services/database/svc/uow.service';
+import { Logger } from '../../services/logging/logging';
+import { Location } from './location';
+
+@Injectable()
+export class LocationDataMapperImpl extends GenericDataMapper
+  implements IDataMapper<Location>, IAclPerm {
+  // ACL permissions
+  public readonly CREATE: string = 'location:create';
+  public readonly DELETE: string = 'location:delete';
+  public readonly READ: string = 'location:read';
+  public readonly UPDATE: string = 'location:update';
+  public readonly READ_ALL: string = 'location:readall';
+  public readonly COUNT: string = 'location:count';
+  public tableName = 'LOCATIONS';
+
+  protected pkColumnName = 'uid';
+
+  constructor(
+    @Inject('IAcl') acl: IAcl,
+    @Inject('MysqlUow') protected readonly sql: MysqlUow,
+    @Inject('BunyanLogger') protected readonly logger: Logger,
+  ) {
+    super(acl);
+    super.addRBAC(
+      [this.CREATE, this.UPDATE, this.DELETE],
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    );
+    super.addRBAC([this.READ, this.READ_ALL], [AuthLevel.PARTICIPANT]);
+  }
+
+  public delete(id: UidType): Promise<IDbResult<void>> {
+    const query = squel
+      .delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, id)
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    )
+      .pipe(map(() => ({ result: 'Success', data: undefined })))
+      .toPromise();
+  }
+
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<Location>> {
+    let queryBuilder = squel
+      .select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder.where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query<Location>(query.text, query.values, {
+        cache: true,
+        stream: false,
+      }),
+    )
+      .pipe(
+        map((location: Location) => ({ result: 'Success', data: location })),
+      )
+      .toPromise();
+  }
+
+  public getAll(): Promise<IDbResult<Stream<Location>>> {
+    const query = squel
+      .select({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName, 'location')
+      .field('location.*')
+      .field('location.location_name')
+      .join('HACKATHON', 'h', 'h.uid=location.hackathon and h.active=true')
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<Location>(query, [], { stream: true, cache: true }),

```

Comment:
Could you perhaps create a variable `params` here similar to the method below for consistency? In case you copied this from somewhere else, please also point that out to me so I can fix that thanks!